server/routes/contacts/create/enter-dob/createContactEnterDobSchemas.ts

```diff
@@ -0,0 +1,43 @@
+import z from 'zod'
+import { createSchema } from '../../../../middleware/validationMiddleware'
+
+const SELECT_IS_DOB_KNOWN_MESSAGE = 'Select whether the date of birth is known'
+const DAY_TYPE_MESSAGE = 'Enter a valid day of the month'
+const MONTH_TYPE_MESSAGE = 'Enter a valid month'
+const YEAR_TYPE_MESSAGE = 'Enter a valid year'
+const DAY_REQUIRED_MESSAGE = 'Enter day'
+const MONTH_REQUIRED_MESSAGE = 'Enter month'
+const YEAR_REQUIRED_MESSAGE = 'Enter year'
+
+export const createContactEnterDobSchema = () => async () => {
+  return createSchema({
+    isDobKnown: z.string({ message: SELECT_IS_DOB_KNOWN_MESSAGE }).transform(value => value === 'true'),
+    day: z.coerce.number({ message: DAY_TYPE_MESSAGE }).optional(),
+    month: z.coerce.number({ message: MONTH_TYPE_MESSAGE }).optional(),
+    year: z.coerce.number({ message: YEAR_TYPE_MESSAGE }).optional(),
+    dob: z.date().optional(),
```

Comment:
Not actually optional but couldn't find a better way of doing this with zod. It's populated via superRefine and transform functions but if you don't populate it here then it's not present in the generated type.

Reply:
I'll need to stare at this for a while .. or read the zod docs..